internal/common/compilation.bzl

```diff
@@ -292,6 +290,36 @@ def compile_ts(ctx,
   if not is_library:
     files += depset(tsickle_externs)
 
+  rerooted_es6_sources = []
+  es6_js_modules_roots = []
+  for source in es6_sources:
+    root = source.dirname
+    es6_js_modules_roots.append(root)
+
+    root_without_prefix = root
+    prefixes_to_remove = [
+        # Bazel < 0.8.0.
+        "bazel-out/local-fastbuild/bin/",
+        # Bazel >= 0.8.0.
+        "bazel-out/k8-fastbuild/bin/",
+    ]
+    for prefix in prefixes_to_remove:
+      if root[0:len(prefix)] == prefix:
+        root_without_prefix = root[len(prefix):]
+
+    file_path = "%s/%s" % (root_without_prefix, source.basename.replace(".closure", ""))
+    print(file_path)
+    rooted_file = ctx.actions.declare_file(file_path)
+
+    ctx.actions.run_shell(
```

Comment:
note, we want to support Windows soon. I am not sure if this will work, it might rely on Posix `cp` command?

Reply:
Donezo